shared_task_code/results_parser.py

```diff
@@ -53,38 +53,47 @@ def process_results(self, topic, bucket, key):
         content = obj["Body"].read()
         tar = tarfile.open(mode="r:gz", fileobj=io.BytesIO(content), format=tarfile.PAX_FORMAT)
         result_file_name = re.sub(r"\.tar.gz$", "", key.split("/", -1)[-1])
-        body = tar.extractfile(result_file_name).read().decode('utf-8').split('\n')
+        body = tar.extractfile(result_file_name).read()
 
-        # we need these to track the scan (TODO: this will be removed and replaced by a better identifier later)
-        start_time, end_time = msgdata["scan_end_time"], msgdata["scan_end_time"]
-        scan_id = os.path.splitext(result_file_name)[0]
+        # there might be times where we want to handle the results ourselves, if so, pass out
+        # the file and let the scanning task handle it themselves, otherwise set up ResultsContext
+        # and pass the body back to the scanning task
 
-        # set up the base set of keys to send to elastic
-        non_temporal_key = {
-            "address": msgdata['address'],
-            "address_type": msgdata['address_type']
-        }
-        # TODO store start_time correctly
-        # TODO end_time will be correct once the scheduler is in place to change the key (currently end_time is used)
+        if self.func_custom_data_handler != None:
+            # TODO: document this
+            self.func_custom_data_handler(body, tar_data=tar, msg_data=msgdata, task_topic=topic, task_bucket=bucket, task_key=key, result_file_name=result_file_name, sns_client=self.sns_client)
+        else:
+            start_time, end_time = msgdata["start_time"], msgdata["end_time"]

```

Comment:
timing doesn't work for ecs tasks

Reply:
it does now - in task_script.sh I've added start_time and end_time as metadata variables that are set in the script - so if you're running something that doesn't provide this then those values can be used